test/app-luatest/gh_7000_compat_module_test.lua

```diff
@@ -0,0 +1,262 @@
+local t = require('luatest')
+local g = t.group()
+local compat = require('tarantool').compat
+
+local NEW = 'new'
+local OLD = 'old'
+local DEFAULT = 'default'
+
+local reset = function(compat)
+    local f = loadstring(compat.dump('default'))
+    f()
+end
+
+g.before_all(function() reset(compat) end)
+g.after_all( function() reset(compat) end)
+
+local option_1_called = false
+local option_2_called = false
+
+local hot_reload_option_called
+
+local policies = {
+    option_1_policy = {
+        name = 'option_1',
+        default = 'new',
+        brief = 'option_1',
+        action = function()
+            option_1_called = true
+        end,
+        run_action_now = true
+    },
+    option_2_policy = {
+        name = 'option_2',
+        default = 'old',
+        brief = 'option_2',
+        obsolete = nil,         -- Explicitly mark option as non-obsolete.
+        action = function()
+            option_2_called = true
+        end,
+        run_action_now = false
+  },
+}
+
+local obsolete_policy = {
+    name = 'obsolete_option',
+    default = 'new',
+    brief = 'obsolete_option',
+    obsolete = '5.0'
+  }
+
+local test_options_calls = function()
+    t.assert(option_1_called)
+    t.assert(option_2_called)
+    option_1_called = false
+    option_2_called = false
+end
+
+g.test_add_option = function()
+    for _, elem in pairs(policies) do
+        compat.add_option(elem)
+    end
+    t.assert_equals(type(compat.option_1), 'table')
+    t.assert_equals(type(compat.option_2), 'table')
+    t.assert(option_1_called)
+    t.assert_not(option_2_called)
+    option_1_called = false
+
+
+    local bad_policy = {                        -- Obsolete option with default == 'old'.
```

Comment:
Nit: Please, follow our usual style: a comment is usually placed on its own line(s). Here nothing prevents it.

Reply:
Fixed.